lib/src/components/NxModal/__tests__/NxModal.test.tsx

```diff
@@ -33,17 +37,102 @@ describe('NxModal', function() {
+        document.body.removeChild(containerMainModal);
+        containerMainModal = null;
+      }
+
+      if (containerSecondaryModal) {
+        document.body.removeChild(containerSecondaryModal);
+        containerSecondaryModal = null;
+      }
+    });
+
+    it('executes onClose method when pressing ESC key', function () {
+      const mockCallBack = jest.fn();
+      const nxModal = <NxModal id="first-modal-id" onClose={mockCallBack}/>;
+
+      act(() => {
+        mount(nxModal, {attachTo: containerMainModal});
+      });
+
+      document.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape'}));
+      expect(mockCallBack).toHaveBeenCalledTimes(1);
+    });
+
+    it('executes onClose method for each modal that has been opened, preserving order call', function () {
+      const firstMockCallBack = jest.fn();
+      const secondMockCallBack = jest.fn();
+      const firstNxModal = <NxModal id="first-modal-id" onClose={firstMockCallBack}/>;
+      const secondNxModal = <NxModal id="second-modal-id" onClose={secondMockCallBack}/>;
+
+      let secondModalWrapper: ReactWrapper | null = null;
+      let firstModalWrapper: ReactWrapper | null = null;
+      act(() => {
+        firstModalWrapper = mount(firstNxModal, {attachTo: containerMainModal});
+        secondModalWrapper = mount(secondNxModal, {attachTo: containerSecondaryModal});
+      });
+
+      // "Close" the second modal
+      document.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape'}));
+      expect(secondMockCallBack).toHaveBeenCalledTimes(1);
+      expect(firstMockCallBack).not.toHaveBeenCalled();
+
+      // Unmount and remount the modal to ensure listener order is preserved if new modals are opened
+      act(() => {
+        secondModalWrapper!.unmount();
+        secondModalWrapper = mount(secondNxModal, {attachTo: containerSecondaryModal});
+      });
+      // "Close" the second modal again
+      document.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape'}));
+      expect(secondMockCallBack).toHaveBeenCalledTimes(2);
+      expect(firstMockCallBack).not.toHaveBeenCalled();
+
+      // Unmount the second modal to remove it's listener from available listeners to execute
+      act(() => {
+        secondModalWrapper!.unmount();
+      });
+      // "Close" the first modal
+      document.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape'}));
+      expect(secondMockCallBack).toHaveBeenCalledTimes(2);
+      expect(firstMockCallBack).toHaveBeenCalledTimes(1);
+
+      // Unmount the first modal to remove it's listener from available listeners to execute
+      act(() => {
+        firstModalWrapper!.unmount();
+      });
+      // Ensure there are no more calls to the registered close handlers with new keypresses
+      document.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape'}));
+      expect(secondMockCallBack).toHaveBeenCalledTimes(2);
+      expect(firstMockCallBack).toHaveBeenCalledTimes(1);
+    });
+  });
```

Comment:
Oh, also test that keypresses other than ESC don't close the modal

Reply:
> Oh, also test that keypresses other than ESC don't close the modal

Test handler is only called with Escape key and not pressing any other common keys: dad801c